include/ola/base/StrError_R.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * StrError_R.h
+ * Helper functions for strerror_r and a declaration of strerror_r that is
+ * XSI-compliant.
+ * Copyright (C) 2018 Shenghao Yang
+ */
+
+/**
+ * @defgroup strerror Description of system error codes
+ * @brief Error descriptions
+ * @details Convenience functions to obtain descriptions of system error codes.
+ * The functions in this group are only defined if \c strerror_r() is available.
+ */
+
+/**
+ * @addtogroup strerror
+ * @{
+ * @file StrError_R.h
+ * @}
+ */
+
+#ifndef INCLUDE_OLA_BASE_STRERROR_R_H_
+#define INCLUDE_OLA_BASE_STRERROR_R_H_
+
+#include <string>
+
+namespace ola {
+
+/**
+ * @addtogroup strerror
+ * @{
+ */
+
+/**
+ * @brief Length of the internal buffer used for @ref StrError_R().
+ * 
+ * If the length of the system-provided error description exceeds the length
+ * of this buffer minus one, then the output of @ref StrError_R() will only
+ * include the numerical error value provided.
+ */
+extern const int StrError_R_BufSize;
```

Comment:
We usually `static const`, is there a reason you need extern?

Reply:
If we `static const` here - then every compilation unit that includes this header file would have their own copy of `StrError_R_BufSize`. `extern` here just lets me allocate storage for it in that one unit and have the other units referencing this variable link to it.

Reply:
We could `#define` stuff.... but I'd want to avoid that route - maybe future refactoring to `constexpr` or `inline constexpr` in future?

Reply:
One for @nomis52 then as to why we've gone for static const over extern const?